lib/topological_inventory/schema/base.rb

```diff
@@ -18,26 +21,28 @@ def to_hash
         }.merge(super)
       end
 
-      def self.klass_for(name)
-        klass_name = "TopologicalInventory::Schema::#{name}"
-        klass = klass_name.safe_constantize
-
-        # sometimes autoloader is confused and loads class with another namespace
-        # checks whether it loads what we want
-        klass if klass.to_s == klass_name
-      end
+      class << self
+        def from_hash(persister_data, manager)
+          persister = super(persister_data, manager)
```

Comment:
If you're passing the same arguments you don't need to explicitly pass them to `super`